bindings/pyroot_experimental/pythonizations/python/ROOT/_numbadeclare.py

```diff
@@ -0,0 +1,278 @@
+        input_signature = ', '.join('{} x_{}'.format(t, i) for i, t in enumerate(input_types_ref))
+
+        # Define function pointer types
+        func_ptr_input_types = []
+        for t in input_types:
+            if 'RVec' in t:
+                innert = get_inner_type(t)
+                if innert == 'bool':
+                    # Special treatment for bool: In numpy, bools have 1 byte
+                    func_ptr_input_types += ['char*, int']
+                else:
+                    func_ptr_input_types += ['{}*, int'.format(innert)]
+            else:
+                func_ptr_input_types += [t]
+        if 'RVec' in return_type:
+            # See C++ wrapper code for the reason using these types
+            innert = get_inner_type(return_type)
+            func_ptr_input_types += ['{}**, long*, int'.format('char' if innert == 'bool' else innert)]
+        func_ptr_type = '{RETURN_TYPE}(*)({INPUT_TYPES})'.format(
+                RETURN_TYPE='void*' if 'RVec' in return_type else return_type,
+                INPUT_TYPES=', '.join(func_ptr_input_types)
+                )
+
+        # Define function call
+        vecbool_conversion = []
+        func_args = []
+        for i, t in enumerate(input_types):
+            if 'RVec' in t:
+                func_args += ['x_{0}.data(), x_{0}.size()'.format(i)]
+                if get_inner_type(t) == 'bool':
+                    # Copy the RVec<bool> to a RVec<char> to match the numpy memory layout
+                    func_args[-1] = func_args[-1].replace('x_', 'xb_')
+                    vecbool_conversion += ['ROOT::RVec<char> xb_{0} = x_{0};'.format(i)]
+            else:
+                func_args += ['x_{}'.format(i)]
+        if 'RVec' in return_type:
+            # See C++ wrapper code for the reason using these arguments
+            func_args += ['&ptr, &size, dtype_size']
+
+        # Define return operation
+        if 'RVec' in return_type:
+            innert = get_inner_type(return_type)
+            if innert == 'bool': innert = 'char'
+            return_op = '\n    '.join([
+                '// Because an RVec cannot take the ownership of external data, we have to copy the returned array',
+                'long size; // Size of the returned array',
+                '{}* ptr; // Pointer to the data of the returned array'.format(innert),
+                'int dtype_size = sizeof({}); // Size of array datatype in bytes'.format(innert),
+                'funcptr({});'.format(', '.join(func_args)),
+                'ROOT::RVec<{}> x_r(size);'.format(innert),
+                'memcpy(x_r.data(), ptr, size * dtype_size);',
+                'free(ptr);',
+                # If we return a RVec<bool>, we rely here on the automatic conversion of RVec<char> to RVec<bool>
+                'return x_r;'])
+        else:
+            return_op = 'return funcptr({});'.format(', '.join(func_args))
+
+        # Build wrapper code
+        cppwrappercode = """\
+#pragma cling optimize(3)
```

Comment:
this does not do anything right?

Reply:
Yes, right. As we learnt in the PPP, the code generation does actually not happen.

Reply:
But it does not hurt :D

Reply:
it's misleading, can you remove it please?

Reply:
Done